### DISCUSSION TITLE
yarn 1 to yarn 4 upgrade

### DISCUSSION THREAD

**Discussion by Justinzobel** [mastodon/mastodon#32250] [General]
I noticed while reading the 4.3.0 upgrade notes to help me get prepared for its release that the yarn version will need changing. Will a step by step guide on how to do this be included in the final 4.3.0 release notes?

All the other dependencies I am capable of managing.

**Comment by renchap** [comment]
Yes, this is included in the 4.3.0 release notes.

**Comment by TonyHoyle** [comment]
Any chance of a link?  There's nothing on https://github.com/mastodon/mastodon/releases/tag/v4.3.0 - just a generic instruction to 'run corepack prepare', which installs yarn 1.22

(I think this is the reason the db:encryption:init fails with a missing file later, because presumably that's a part of yarn4).

**Reply by renchap** [reply]
If you ran `corepack prepare`, then `corepack enable` once you checked out Mastodon's 4.3.0 code should install yarn. I think that simply running `yarn install` will also install the correct version.

This is not related to your `db:encryption:init` error, so you will need to provide more details if you need help on this.

**Comment by Takumiboo** [comment]
In my server, I ran`corepack prepare` and `corepack enable` and installed yarn 4.5.0, but server used yarn 1.22 when runnnig `yarn install`.
I ran `yarn set version 4.5.0` and going well, but I don't know why.

**Comment by marinhio75** [comment]
RAILS_ENV=production bundle exec rails assets:precompile
Webpacker requires Yarn ">=1 <4" and you are using 4.5.0
This version of Webpacker does not support Yarn 4.5.0. Please downgrade to a supported version of Yarn https://yarnpkg.com/lang/en/docs/install/
Exiting!

Lol how funny Mastodon needs Yarn 4, gets Yarn 4 and wants a downgrade?

Mastodon recommends using corepack from NodeJS and NodeJS recommends using external package managers and removes corepack.

Confusing...

**Comment by thomasmielke70** [comment]
Any new or more detailed information about the Yarn version or dependencies? I didn't manage to upgrade my instance yet. I've tried to upgrade several times and was stuck every time with "Webpacker requires Yarn ">=1 <4" and you are using 4.5.0". I had to roll back to 4.2.13 to ensure my instance is running.
